### DISCUSSION TITLE
Filament admin running extremely slow

### DISCUSSION THREAD

**Discussion by faridsa** [filamentphp/filament#5654] [Help] [answered]
Hello, I am building an administration panel with a standard installation of filament 2.16.70 using a fresh laravel 9.51.
Following the docs I have been able to create the necessary resources and forms to manage different data models. 
The problem I'm noticing is that the entire admin area is running extremely slow and I would like to know if there are any setting that I'm omitting to cause that.

I'm trying disabling Laravel Debug Bar, but there is no difference.
![Screenshot 2023-02-12 at 15 03 36](https://user-images.githubusercontent.com/23212965/218330199-f9612041-4373-4b8b-9026-2ea71fc39f12.jpeg)

**Comment by danharrin** [comment]
563 queries and 7851 models is why its running slow...

**Reply by danharrin** [reply]
you'll need to share code where you're doing that

**Reply by faridsa** [reply]
Thank you for your reply Dan, I wish to know but why filament makes that queries?
I can't understand that situation, I've generated resource class as stated in docs using `php artisan make:filament-resource Setting` and that's all.

My resource class is the following

```
namespace App\Filament\Resources;

use App\Filament\Resources\SettingResource\Pages;
use App\Filament\Resources\SettingResource\RelationManagers;
use App\Models\Setting;
use Filament\Forms;
use Filament\Resources\Form;
use Filament\Resources\Resource;
use Filament\Resources\Table;
use Filament\Tables;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\SoftDeletingScope;

class SettingResource extends Resource
{
    protected static ?string $model = Setting::class;

    protected static ?string $navigationIcon = 'heroicon-o-cog';
    protected static ?string $modelLabel = 'Dato';
    protected static ?int $navigationSort = 5;

    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\TextInput::make('name')->label('Nombre')->extraInputAttributes(['readonly' => true]),
                Forms\Components\TextArea::make('value')->label('Valor')->required(),
            ]);
    }

    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('name')->label('Nombre'),
                Tables\Columns\TextColumn::make('value')->label('Valor'),
            ])
            ->filters([
                //
            ])
            ->actions([
                Tables\Actions\ViewAction::make(),
                Tables\Actions\EditAction::make(),
            ])
            ->bulkActions([
            ]);
    }

    public static function getRelations(): array
    {
        return [
        ];
    }

    public static function getPages(): array
    {
        return [
            'index' => Pages\ListSettings::route('/'),
            'view' => Pages\ViewSetting::route('/{record}'),
            'edit' => Pages\EditSetting::route('/{record}/edit'),
        ];
    }

    public static function canCreate(): bool
   {
      return false;
   }

}
```

**Comment by danharrin** [comment, marked as answer]
Please open the Queries tab and screenshot a wide range of queries that are being run so I can see where they come from

**Reply by danharrin** [reply]
Or, you can send the repository with setup instructions

**Reply by faridsa** [reply]
Thank you Dan so much. You guide me in the right direction. 
It was due to values added to boot function in AppServiceProvider in order to share with public views.

```
view()->composer('*', function ($view) {
            $settingsArr = \App\Models\Setting::all()->toArray();
            $settings = Arr::pluck($settingsArr, 'value', 'name');
            $cart['items'] =  \Cart::isEmpty() ? null : \Cart::getContent();
            $view->with(['settings'=>$settings, 'cart' => $cart]);
        });  
```

Whe commenting out that block, everythings is running smoothly now.

**Reply by faridsa** [reply]
I dif a further investigation in order to share variables only with needed views, and now my AppServiceProvider boot function is just like:
```
view()->composer('public.*', function ($view) {
            $settingsArr = \App\Models\Setting::all()->toArray();
            $settings = Arr::pluck($settingsArr, 'value', 'name');
            $cart =  \Cart::isEmpty() ? null : \Cart::getTotalQuantity();
            $view->with(['settings'=>$settings, 'cart' => $cart]);
        });  
```
With that change, everything is working as expected.

One more time, thank you so much for your assistance.

**Reply by bert-w** [reply]
> I dif a further investigation in order to share variables only with needed views, and now my AppServiceProvider boot function is just like:
> 
> ```
> view()->composer('public.*', function ($view) {
>             $settingsArr = \App\Models\Setting::all()->toArray();
>             $settings = Arr::pluck($settingsArr, 'value', 'name');
>             $cart =  \Cart::isEmpty() ? null : \Cart::getTotalQuantity();
>             $view->with(['settings'=>$settings, 'cart' => $cart]);
>         });  
> ```
> 
> With that change, everything is working as expected.
> 
> One more time, thank you so much for your assistance.

Randomly came across this thread and saw this code:

You MUST change this code to `Setting::query()->pluck('value', 'name');`. The way you are creating the models is absolutely devastating, but doing a pluck on the query builder will create the array you want with `'name'` as the key and `'value'` as the value. This bypasses the whole model creation.

**Comment by christoferd** [comment]
I have reached 1590 views in a single request.
![image](https://user-images.githubusercontent.com/1835883/229906931-474906d1-b81d-4461-88e1-63c8b3fa5bcd.png)

**Reply by danharrin** [reply]
View count means nothing. We are using Blade components to eliminate code duplication which is why it is high. Think of them as simple PHP includes.

Disable the view collector for debugbar as that will slow the app down lots.

**Reply by sneycampos** [reply]
I was facing the same "slow problem" and yes, was view collector :D

**Reply by amadeann** [reply]
The difference is massive, with the view collector the app was almost unusable. Glad I found this comment :)

**Reply by ralphmorris** [reply]
Massive help thank you!

**Comment by kedniko** [comment]
I solved it by disabling Xdebug on every request.

In `php.ini` i commented this option
`// xdebug.start_with_request="yes"`
and replaced it with
`xdebug.start_with_request=trigger`

...then in my code i use `xdebug_break()` when needed

I'm using xdebug 3, check for documentation for version 2

**Comment by catative** [comment]
I also encountered this problem. On a completely clean installation of Laravel 10.40.0 and Filament 3.1.42, I created one simple model, one migration for it with one custom text field and one Filament resource. Going (in UI) to the section of the created resource or using any action, for example, creating a new model record in DB, took a very long time (about 10 seconds). Disabling the Laravel debug panel (barryvdh/laravel-debugbar) or xDebug did not result in success. The problem was on both the Apache and Nginx servers (on Ubuntu 22.04).

Experiencedly, I noticed that this problem began to occur with Filament 3.0.28 version (in 3.0.27 everything worked quickly). It turned out that in my case the problem was due to the **Adblock Plus extension** for the Google Chrome browser. When I turned it off, everything immediately began to work quickly even in the latest versions of Filament and even with enabled xDebug.

**Reply by adman9000** [reply]
Plus one for disabling adblock. I had to disable it fully, pausing it on the site only didn't help.

**Reply by shankhadevpadam** [reply]
I face the same issue. Filament runs very slow when enabled ad block extension.

**Reply by jratajski** [reply]
Thank you, I had the same problem with the livewire/update request that was pending for 6-7 seconds before it was started.

**Reply by codifyer** [reply]
Faced same issue. I recently tried out Filament, installed v3.2.52 on Laravel 11 locally (Windows 11 Ryzen7 16GB RAM), used Chrome with no ad blockers, and the screen loads are painfully slow, that too with just a few rows of test data !! Something is not right.

**Reply by danharrin** [reply]
If you're on WSL, check this thread - https://github.com/filamentphp/filament/discussions/6120#discussioncomment-10661574

**Comment by medfrk** [comment]
I am experiencing a similar problem. In my local environment, everything works fine. However, when I deployed my project on a new VPS that I have configured, which boasts an impressive setup of 8GB of RAM and a 4-core CPU, I faced issues. Unfortunately, the app was very slow. This sluggishness is due to Filament, as every time we modify an element in the form, it makes a request to the server. I am unsure if it's possible to implement some data binding on the client side to avoid requesting the server every time.

**Reply by danharrin** [reply]
Forms are not live by default. Are you using `live()` everywhere?

**Reply by medfrk** [reply]
Yes, I am using live(), and that's precisely the problem. live() triggers a server request every time a change is made, which is causing the performance issues I mentioned.

**Reply by sneycampos** [reply]
Your database is in the same server or network?

**Reply by medfrk** [reply]
The database is hosted on the same server as the application

**Reply by danharrin** [reply]
`live()` is for dynamic forms, is every single form in your app reactive? Have you tried Octane?

**Reply by medfrk** [reply]
Indeed, every form in the app is reactive, utilizing live(). I haven't tried Octane yet, but it seems like a potential solution. I will look into it. Thank you for the suggestion!

**Reply by medfrk** [reply]
Is there a possibility to integrate a mechanism in FilamentPHP to have a functionality similar to React? I'm interested in reactive features to enhance the user experience.

**Reply by danharrin** [reply]
Please really think carefully if every form really needs to be reactive, that is very heavy. There is no option to change the frontend framework for Filament, that would not be maintainable for us.

**Comment by shankhadevpadam** [comment]
@danharrin there is an issue with ad block extension in Chrome. If I enable ad block extension the filament runs very slow as compared to disabling ad block extension or private mode.

**Reply by danharrin** [reply]
I know, there have been many reports. I do not know what Filament can do, its not really our problem, its the extension fighting with Chrome when loading JavaScript.

**Comment by sasaorasanin** [comment]
Also it can be a chrome problem, in last 15 days I got most websites very slow, while on firefox it worked well. Also on android chrome works well.

**Reply by sneycampos** [reply]
i am using chrome and my apps looks normal and with ublock origin.

Are you using any adblock?

**Comment by ivl04** [comment]
"I've cached icons, disabled the debugger bar, removed unnecessary packages, and performed a fresh Composer installation after deleting the 'vendor' directory and 'composer.lock' file. Additionally, I've disabled the Zend extension in the 'php.ini' file. Despite these actions, I'm still encountering the issue. Any suggestions on how to proceed?"

**Reply by ratajski** [reply]
Have you tried a different browser, especially one without any extensions. Its a common problem that adblockers delay the ajax requests.

**Reply by solivellaluisalberto** [reply]
Same problem here. Running all possible optimization commands, working in a browser with no extensions and with all php extensions I've read about disabled. When selecting 100 records in a table, the application does not work as it should.

**Reply by christoferd** [reply]
I have another thread on this. 
1. Try disabling the Zend Extension X-Debug via your php.ini
2. Add web project's root folder to Windows Defender Exclusions list
- if you have any other anti-malware or anti-virus apps running, then also do it in those apps

PS: Personally, I keep all my dev stuff under C:\www and just add that whole folder to Windows Defender Exclusions list

**Comment by sasaorasanin** [comment]
Guys there is one more thing that I had, for example by  using herd on windows and generating new app with class livewire out of the box and then adding filament. For some strange reason app is so slow I get timeout every single time. But if I generate new app without livewire or any starter kit, then install livewire and filament, everything is smooth. And not just locally but on server also.
